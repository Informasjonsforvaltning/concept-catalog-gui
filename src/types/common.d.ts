import { RouteComponentProps } from 'react-router';

export type RouterProps = RouteComponentProps<{ catalogId?: string; conceptId?: string }>;
