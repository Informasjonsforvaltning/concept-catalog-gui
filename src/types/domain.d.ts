export interface ErrorMessage {
  name?: string;
  message?: string;
}

export interface ImportErrorMessage extends ErrorMessage {
  thrown: boolean;
}

export interface InvalidConceptErrorMessage {
  index: number;
  message: string;
  conceptTitle?: string;
}

export interface Kilde {
  uri: string;
  tekst: string;
}

export interface Language {
  code: string;
  title: string;
  selected: boolean;
}

export interface ListItem {
  title: string;
  theme: string;
  valid: string;
  status: string;
}

export default class ImportError {
  constructor(message: string, name?: string) {
    const error = Error(message);

    Object.defineProperty(error, 'message', {
      get() {
        return message;
      }
    });
    Object.defineProperty(error, 'name', {
      get() {
        return name ?? 'ImportError';
      }
    });
    return error;
  }
}

export interface UriText {
  uri?: string;
  tekst?: string;
}

export interface ContactDetails {
  harEpost?: string;
  harTelefon?: string;
}

export interface AnbefaltTerm {
  navn: TekstMedSpraakKode;
}

export interface Definisjon {
  tekst: TekstMedSpraakKode;
}

export interface TekstMedSpraakKode {
  [kode: string]: string;
}

export interface Navn {
  nb?: any;
  nn?: any;
  en?: any;
}

export interface Version {
  major: number;
  minor: number;
  patch: number;
}

export interface Concept {
  id: string;
  originaltBegrep?: string;
  versjonsnr?: Version | null;
  revisjonAv?: string;
  anbefaltTerm?: AnbefaltTerm;
  definisjon?: Definisjon;
  ansvarligVirksomhet: { id: string };
  kildebeskrivelse?: {
    forholdTilKilde: string;
    kilde: Kilde[];
  } | null;
  merknad?: Record<string, string[]>;
  eksempel?: Record<string, string[]>;
  fagomr책de?: TekstMedSpraakKode;
  bruksomr책de?: Record<string, string[]>;
  omfang?: UriText | null;
  tillattTerm?: Record<string, string[]>;
  frar책detTerm?: Record<string, string[]>;
  kontaktpunkt?: ContactDetails | null;
  gyldigFom?: string | null;
  gyldigTom?: string | null;
  seOgs책: string[];
  status?: string | null;
  erSistPublisert?: boolean;
  revisjonAvSistPublisert?: boolean;
}
