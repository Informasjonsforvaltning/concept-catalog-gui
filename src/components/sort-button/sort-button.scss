.sortButton {
  background-color: transparent;
  border: 0;
  padding: 0 0.5em;
}
