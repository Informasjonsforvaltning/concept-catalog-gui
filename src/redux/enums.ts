export enum CommentReducer {
  name = 'comments'
}
