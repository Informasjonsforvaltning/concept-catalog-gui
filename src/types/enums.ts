export enum SortDirection {
  ASC = 'asc',
  DESC = 'desc'
}

export enum ConceptStatus {
  UTKAST = 'utkast',
  PUBLISERT = 'publisert'
}

export enum ConceptField {
  TERM = 'anbefaltTerm.navn.nb',
  FAGOMRÅDE = 'fagområde',
  VALID = 'valid',
  VERSION = 'versjonsnr',
  MODIFY_TIME = 'endringslogelement.endringstidspunkt',
  STATUS = 'status'
}


export enum TimeFormat {
  dateAndHour = 'DDDD kl. HH:mm'
}
