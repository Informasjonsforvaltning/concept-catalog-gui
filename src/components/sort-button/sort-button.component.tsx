import React from 'react';

import { SortDirection } from '../../types/enums';
import './sort-button.scss';

interface Props {
  field: string;
  sortField: string;
  sortType: SortDirection;
  onSortField: (field: string, sortDirection: SortDirection) => void;
}

export const SortButtons = ({
  field,
  sortField,
  sortType,
  onSortField
}: Props): JSX.Element | null => (
  <div className='d-flex flex-column'>
    <button
      type='button'
      name={`${field}asc`}
      className={`d-flex sortButton ${
        sortField === `${field}` && sortType === SortDirection.ASC
          ? 'visibilityHidden'
          : ''
      }`}
      onClick={(): void => onSortField(`${field}`, SortDirection.ASC)}
      title='Stigende'
    >
      <i className='fa fa-sort-up' />
    </button>
    <button
      type='button'
      name={`${field}desc`}
      className={`d-flex sortButton ${
        sortField === `${field}` && sortType === 'desc'
          ? 'visibilityHidden'
          : ''
      }`}
      onClick={(): void => onSortField(`${field}`, SortDirection.DESC)}
      title='Synkende'
    >
      <i className='fa fa-sort-down' />
    </button>
  </div>
);
