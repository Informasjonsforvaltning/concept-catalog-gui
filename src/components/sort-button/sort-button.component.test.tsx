import React from 'react';
import { shallow } from 'enzyme';
import { SortButtons } from './sort-button.component';
import { SortDirection } from '../../types/enums';

test('should render SortButtons', (): void => {
  const onSortField = jest.fn();
  const defaultProps = {
    field: 'title',
    sortField: 'sortField',
    sortType: SortDirection.ASC,
    onSortField
  };
  const wrapper = shallow(<SortButtons {...defaultProps} />);
  expect(wrapper).toMatchSnapshot();
});
