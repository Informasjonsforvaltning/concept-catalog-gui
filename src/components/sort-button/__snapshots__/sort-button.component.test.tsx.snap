// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SortButtons 1`] = `
<div
  className="d-flex flex-column"
>
  <button
    className="d-flex sortButton "
    name="titleasc"
    onClick={[Function]}
    title="Stigende"
    type="button"
  >
    <i
      className="fa fa-sort-up"
    />
  </button>
  <button
    className="d-flex sortButton "
    name="titledesc"
    onClick={[Function]}
    title="Synkende"
    type="button"
  >
    <i
      className="fa fa-sort-down"
    />
  </button>
</div>
`;
